### PR TITLE
Format record braces consistently with class

### LINKED ISSUE #233: Record - does not format consistently with a class.
Records should format the same as classes. This is how a record currently formats.
```c#
    public class AddResultParameter
    {
        public Guid ActionRequestId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Value { get; set; } = string.Empty;
    }
// vs
    public record AddResultParameter {
        public Guid ActionRequestId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Value { get; set; } = string.Empty;
    }
```

### PR DESCRIPTION
**Pull request by belav**
closes #233

### CODE REVIEW

**Review by shocklateboy92 — APPROVED** [review]

File: Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/BaseTypeDeclaration.cs [hunk 1]
```diff
@@ -105,7 +105,9 @@ public static Doc Print(BaseTypeDeclarationSyntax node)
                 DocUtilities.RemoveInitialDoubleHardLine(members);
 
                 docs.Add(
-                    groupId != null ? Doc.IfBreak(" ", Doc.Line, groupId) : Doc.HardLine,
+                    groupId != null && parameterList != null
```
Comment (shocklateboy92) [inline Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/BaseTypeDeclaration.cs:108]: Relatively minor comment, but it took me a while to understand you're doing this because we want to maintain space-brace when there is a parameter list.
Also the fact that you're using an `IfBreak` here because you want the opposite effect of `Line` - i.e. space if the group does break, line otherwise.

Maybe adding a comment here might make that easier to parse.

**Review by shocklateboy92 — APPROVED** [review]
For full disclosure, this might have been too early in the morning for me.
Add the comment I suggested if you think people will have the same amount of trouble at a reasonable time of the day :coffee:

File: Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/BaseTypeDeclaration.cs (hunk 1 above)
Comment (belav) [reply Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/BaseTypeDeclaration.cs:108]: I added a variable to help explain it. I kind of prefer that to adding comments. I think it forces someone to pay attention if they start changing code, where comments tend to get forgotten and out of date sometimes.

I think `Doc.IfBreak` is hard to parse in general, even though I've used it quite a bit at this point.
Theoretically I think something like this could be implemented, but I'm debating if it is worth the extra effort. Also it may be less obvious how to use, but more obvious what it does when you read it later.
```c#
Doc.IfGroupBreaks(" ").Else(Doc.Line)
// which could be extended to include the groupId somehow
Doc.IfGroup(groupId).Breaks(" ").Else(Doc.Line)
```

Comment (shocklateboy92) [reply Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/BaseTypeDeclaration.cs]: I don't think the method chain would increase readability enough to be worth the cost of implementing it.
Thinking about it some more, using named parameters would probably be enough (assuming the parameters clear names).

Comment (belav) [reply Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/BaseTypeDeclaration.cs]: But what if it is fun to implement??! I threw it together this morning to see what it might look like.
